### PR TITLE
Tacho : SpMV option for Chol & LU

### PR DESCRIPTION
**Pull request by iyamazaki**
@trilinos/shylu 

## Motivation

This PR adds an option to use SpMV for Chol & LU solve

## Testing

Tested using Sierra-SD matrices on V100 and Mi250x

### CODE REVIEW

**Review by crdohrm — DISMISSED** [review]

File: packages/shylu/shylu_node/tacho/src/impl/Tacho_SupernodeInfo.hpp [hunk 1]
```diff
@@ -20,6 +20,13 @@ Sandia National Laboratories, Albuquerque, NM, USA
 #define __TACHO_SUPERNODE_INFO_HPP__
 
 #include "Tacho_Util.hpp"
+#if defined(KOKKOS_ENABLE_CUDA)
```
Comment (ndellingwood) [inline packages/shylu/shylu_node/tacho/src/impl/Tacho_SupernodeInfo.hpp:23]: I'm not sure this is sufficient to ensure cusparse is available? Would be helpful to add more conditions to the check though I'm not sure what is sufficient (cusparse might be available by enabling as a tpl, or through kokkos-kernels or both); would something like:

```c++
#if defined(KOKKOS_ENABLE_CUDA) && (defined(TPL_ENABLE_CUSPARSE) || defined(KOKKOSKERNELS_ENABLE_TPL_CUSPARSE))
```
cover either case?

File: packages/shylu/shylu_node/tacho/src/impl/Tacho_SupernodeInfo.hpp [hunk 2]
```diff
@@ -118,6 +128,26 @@ template <typename ValueType, typename DeviceType> struct SupernodeInfo {
...
+    rowptr_view rowptrU;
+    colind_view colindU;
+    nzvals_view nzvalsU;
+
+    rowptr_view rowptrL;
+    colind_view colindL;
+    nzvals_view nzvalsL;
+
+    bool spmv_explicit_transpose;
+#if defined(KOKKOS_ENABLE_CUDA)
```
Comment (ndellingwood) [inline packages/shylu/shylu_node/tacho/src/impl/Tacho_SupernodeInfo.hpp:141]: Similar to above, 

I'm not sure this is sufficient to ensure cusparse is available? Would be helpful to add more conditions to the check though I'm not sure what is sufficient (cusparse might be available by enabling as a tpl, or through kokkos-kernels or both); would something like:

```c++
#if defined(KOKKOS_ENABLE_CUDA) && (defined(TPL_ENABLE_CUSPARSE) || defined(KOKKOSKERNELS_ENABLE_TPL_CUSPARSE))
```
cover either case?

**Review by ndellingwood — APPROVED** [review]
Changes look reasonable, thanks @iyamazaki

File: packages/shylu/shylu_node/tacho/cmake/Tacho_config.h.in [hunk 3]
```diff
@@ -34,6 +34,9 @@ Sandia National Laboratories, Albuquerque, NM, USA
 /* Define if want to build with BLAS enabled */
 #cmakedefine TACHO_HAVE_BLAS
 
+/* Define if want to build with CuSparse enabled */
+#cmakedefine TACHO_HAVE_CUSPARSE
```
Comment (ndellingwood) [inline packages/shylu/shylu_node/tacho/cmake/Tacho_config.h.in:38]: Nice addition, thanks!

**Review by ndellingwood — APPROVED** [review]

**Comment by iyamazaki** [comment]
Thank you, @ndellingwood!!

**Pull request merged**